### ISSUE TITLE
Phantom developmental community code: "amuse.support.exceptions.CodeException: lost connection to code"

### ISSUE THREAD

**Issue by gautam-404** [amusecode/amuse#864]
**Describe the bug**
Phantom community code does not execute. The exception thrown is "amuse.support.exceptions.CodeException: lost connection to code"

**To Reproduce**
Run example/syllabus/hydro_minimal.py using Phantom instead of Gadget2.
Note: the code works fine with Gadget2

**Logs**
```
Traceback (most recent call last):
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/core.py", line 120, in __call__
    dtype_to_result = self.interface.channel.recv_message(call_id, self.specification.id, handle_as_array)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/channel.py", line 1971, in recv_message
    message.receive(self.socket)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/channel.py", line 1519, in receive
    header_bytes = self._receive_all(44, socket)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/channel.py", line 1504, in _receive_all
    raise exceptions.CodeException("lost connection to code")
amuse.support.exceptions.CodeException: lost connection to code

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "hydro.py", line 45, in <module>
    main(**o.__dict__)
  File "hydro.py", line 14, in main
    Etot_init = hydro.kinetic_energy + hydro.potential_energy + hydro.thermal_energy
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/interface.py", line 1623, in __getattr__
    result = handler.get_attribute(name, result)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/interface.py", line 887, in get_attribute
    return self.property_definitions[name].get_value(value)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/interface.py", line 1681, in get_value
    result = self._method (**self.keyword_arguments)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/methods.py", line 168, in __call__
    result = self.method(*list_arguments, **keyword_arguments)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/methods.py", line 166, in __call__
    object = self.precall()
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/methods.py", line 215, in precall
    return self.definition.precall(self)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/interface.py", line 355, in precall
    transition.do()
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/state.py", line 123, in do
    self.method.new_method()()
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/methods.py", line 168, in __call__
    result = self.method(*list_arguments, **keyword_arguments)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/methods.py", line 168, in __call__
    result = self.method(*list_arguments, **keyword_arguments)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/methods.py", line 168, in __call__
    result = self.method(*list_arguments, **keyword_arguments)
  [Previous line repeated 1 more time]
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/support/methods.py", line 267, in __call__
    return self.method(*list_arguments, **keyword_arguments)
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/core.py", line 123, in __call__
    raise exceptions.CodeException("Exception when calling function '{0}', of code '{1}', exception was '{2}'".format(self.specification.name, type(self.interface).__name__, ex))
amuse.support.exceptions.CodeException: Exception when calling function 'commit_particles', of code 'PhantomInterface', exception was 'lost connection to code'
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
Exception ignored in: <function CodeInterface.__del__ at 0x7f3a90a0a670>
Traceback (most recent call last):
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/core.py", line 749, in __del__
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/core.py", line 839, in _stop
  File "/home/gitpod/.pyenv/versions/3.8.13/lib/python3.8/site-packages/amuse/rfi/core.py", line 123, in __call__
amuse.support.exceptions.CodeException: Exception when calling function '_stop_worker', of code 'PhantomInterface', exception was '[Errno 32] Broken pipe'
```

**Environment (please complete the following information):**
 - OS and version: Ubuntu 20.04.4 LTS
 - Compiler: gcc9

**Comment by rieder** [comment]
When I run it with `hydro = Phantom(converter, redirection='none')`, Phantom shows me a density convergence error on which it crashes.
This does not seem like a bug, rather a problem with the combination of initial conditions and Phantom.
Phantom currently doesn't like it when particles don't show an initial guess of the smoothing length, please could you try setting gas.h_smooth to something slightly sensible (e.g. Rvir / sqrt(N) would be good enough probably) before adding the gas to Phantom?
Also please set the initial timestep (`hydro.parameters.time_step`) to something suitably small.

If this works, you will next find another error - the `thermal_energy` attribute is not yet implemented in Phantom.

In short, Phantom does not (yet?) work as a drop-in replacement for other SPH codes - it requires careful setting up. 
For this particular example, the settings described in https://github.com/danieljprice/phantom/blob/master/src/setup/set_star.f90 might be helpful.

**Comment by gautam-404** [comment]
I appreciate your reply. As I am under some time constraints, I went ahead and started working with the Phantom source code instead. 
I'll try what you recommended over the weekend. Thank you.
